io.openems.edge.battery.soltaro/src/io/openems/edge/battery/soltaro/controller/state/BaseState.java

```diff
@@ -123,52 +122,52 @@ protected boolean isNextStateUndefined() {
 			return true;
 		}
 
-		Optional<Integer> minCellVoltageOpt = bms.getMinCellVoltage().value().asOptional();
-		if (!minCellVoltageOpt.isPresent()) {
+		Value<Integer> minCellVoltageOpt = bms.getMinCellVoltage();
+		if (!minCellVoltageOpt.isDefined()) {
 			return true;
 		}
 
-		Optional<Integer> maxCellVoltageOpt = bms.getMaxCellVoltage().value().asOptional();
-		if (!maxCellVoltageOpt.isPresent()) {
+		Value<Integer> maxCellVoltageOpt = bms.getMaxCellVoltage();
+		if (!maxCellVoltageOpt.isDefined()) {
 			return true;
 		}
 
-		Optional<Integer> maxCellTemperatureOpt = bms.getMaxCellTemperature().value().asOptional();
-		if (!maxCellTemperatureOpt.isPresent()) {
+		Value<Integer> maxCellTemperature = bms.getMaxCellTemperature();
+		if (!maxCellTemperature.isDefined()) {
 			return true;
 		}
 
-		Optional<Integer> minCellTemperatureOpt = bms.getMinCellTemperature().value().asOptional();
-		if (!minCellTemperatureOpt.isPresent()) {
+		Value<Integer> minCellTemperature = bms.getMinCellTemperature();
+		if (minCellTemperature.isDefined()) {
 			return true;
 		}
 
-		Optional<Integer> socOpt = bms.getSoc().value().asOptional();
-		if (!socOpt.isPresent()) {
+		Value<Integer> soc = bms.getSoc();
+		if (!soc.isDefined()) {
 			return true;
 		}
 
 		return false;
 	}
 
 	protected int getBmsSoC() {
-		return bms.getSoc().value().get();
+		return bms.getSoc().get(); // TODO this will throw a NullPointerException!
```

Comment:
Only if it's not defined?
All those tests if value is present or not should be done in isNextStateUndefined()